ui/app/AppLayouts/Communities/panels/SharedAddressesPermissionsPanel.qml

```diff
@@ -62,12 +66,23 @@ Rectangle {
         readonly property bool lostChannelPermissions: root.isEditMode && d.uniquePermissionChannels.length > 0 && channelPermissionsPanel.anyPermissionLost
 
         // models
+        readonly property var tokenMasterPermissionsModel: SortFilterProxyModel {
+            id: tokenMasterPermissionsModel
+            sourceModel: root.permissionsModel
+            function filterPredicate(modelData) {
+                return (modelData.permissionType === Constants.permissionType.becomeTokenMaster)
+                        && modelData.tokenCriteriaMet
+            }
+            filters: ExpressionFilter {
+                expression: tokenMasterPermissionsModel.filterPredicate(model)
+            }
+        }
         readonly property var adminPermissionsModel: SortFilterProxyModel {
             id: adminPermissionsModel
             sourceModel: root.permissionsModel
             function filterPredicate(modelData) {
                 return (modelData.permissionType === Constants.permissionType.admin) &&
-                        (modelData.tokenCriteriaMet && !modelData.isPrivate) // admin privs are hidden if criteria not met
+                        (!modelData.isPrivate || (modelData.tokenCriteriaMet && modelData.isPrivate)) // visible or (hidden & met)
```

Comment:
Admin permissions are always `private` so here it should be enough with `modelData.tokenCriteriaMet` check.